test/e2e/tests/block_test.go

```diff
@@ -55,14 +57,41 @@ func TestBlock_Header(t *testing.T) {
 	})
 }
 
+// Tests that the node configured to prune are actually pruning.
+func TestBlock_Pruning(t *testing.T) {
+	t.Helper()
+	testNode(t, func(t *testing.T, node e2e.Node) {
+		t.Helper()
+		// We do not run this test on seed nodes or nodes with data
+		// companion-related pruning enabled or nodes not pruning.
+		if node.Mode == e2e.ModeSeed || node.EnableCompanionPruning || node.RetainBlocks == 0 {
+			return
+		}
+		client, err := node.Client()
+		require.NoError(t, err)
+		status, err := client.Status(ctx)
+		first0 := status.SyncInfo.EarliestBlockHeight
+		require.NoError(t, err)
```

Comment:
doesn't change the logic, more semantics to test right after the `err` return
```suggestion
		require.NoError(t, err)
		first0 := status.SyncInfo.EarliestBlockHeight
```

Reply:
It's actually a bug. It will panic if it returns an error. Thanks for spotting!